Failing test in test_source_space.py

`nosetests mne/tests/test_source_space.py:test_other_volume_source_spaces` fails for me locally with the current master. Can someone confirm?

it does not pass for me either:

F
======================================================================
FAIL: Test setting up other volume source spaces.
----------------------------------------------------------------------
Traceback (most recent call last):
  File
"/Users/alex/anaconda/envs/py35/lib/python3.5/site-packages/nose/case.py",
line 198, in runTest
    self.test(*self.arg)
  File
"/Users/alex/anaconda/envs/py35/lib/python3.5/site-packages/numpy/testing/decorators.py",
line 147, in skipper_func
    return f(*args, **kwargs)
  File "/Users/alex/work/src/mne-python/mne/utils.py", line 843, in dec
    return function(*args, **kwargs)
  File "/Users/alex/work/src/mne-python/mne/tests/test_source_space.py",
line 257, in test_other_volume_source_spaces
    _compare_source_spaces(src, src_new, mode='approx')
  File "/Users/alex/work/src/mne-python/mne/source_space.py", line 2680, in
_compare_source_spaces
    assert_equal(s0[name], s1[name], name)
AssertionError: 8407 != 7497 : nuse
    '8407 != 7497' = '%s != %s' % _common_shorten_repr(8407, 7497)
    '8407 != 7497 : nuse' = self._formatMessage('8407 != 7497 : nuse',
'8407 != 7497')
>>  raise self.failureException('8407 != 7497 : nuse')


----------------------------------------------------------------------

do you have the same error?

--- a/.travis.yml
+++ b/.travis.yml
@@ -80,7 +80,7 @@ matrix:
 before_install:
     - git clone https://github.com/astropy/ci-helpers.git
     - source ci-helpers/travis/setup_conda.sh
-    - if [ "${DEPS}" == "" && "${TRAVIS_OS_NAME}" == "linux" ]; then
+    - if [ "${DEPS}" == "" ] && [ "${TRAVIS_OS_NAME}" == "linux" ]; then
         curl https://staff.washington.edu/larsoner/minimal_cmds.tar.gz | tar xz;
         export MNE_ROOT="${PWD}/minimal_cmds";
         export NEUROMAG2FT_ROOT="${PWD}/minimal_cmds/bin";

--- a/.travis.yml
+++ b/.travis.yml
@@ -85,7 +85,7 @@ before_install:
         curl https://staff.washington.edu/larsoner/minimal_cmds.tar.gz | tar xz;
         export MNE_ROOT="${PWD}/minimal_cmds";
         export NEUROMAG2FT_ROOT="${PWD}/minimal_cmds/bin";
-        source ${MNE_ROOT}/bin/mne_setup_sh | tee;
+        source ${MNE_ROOT}/bin/mne_setup_sh | tee foo.txt;
       fi;
     - if [ "${TRAVIS_OS_NAME}" == "linux" ]; then
         /sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1400x900x24 -ac +extension GLX +render -noreset;
@@ -150,7 +150,7 @@ script:
         USE_DIRS="mne/";
       fi;
     - echo Testing with -a "${CONDITION}" ${COVERAGE} ${USE_DIRS}
-    - nosetests -a "${CONDITION}" --with-timer --timer-top-n 30 --timer-ok 5 --timer-warning 30 --verbosity=2 ${COVERAGE} ${USE_DIRS};
+    - nosetests -a "${CONDITION}" -x --with-timer --timer-top-n 30 --timer-ok 5 --timer-warning 30 --verbosity=2 ${COVERAGE} ${USE_DIRS};
     - if [ "${DEPS}" == "nodata" ]; then
         make pep;
         check-manifest --ignore doc,logo,mne/io/*/tests/data*,mne/io/tests/data,mne/preprocessing/tests/data;

--- a/.travis.yml
+++ b/.travis.yml
@@ -80,7 +80,7 @@ matrix:
 before_install:
     - git clone https://github.com/astropy/ci-helpers.git
     - source ci-helpers/travis/setup_conda.sh
-    # Pipe the source into tee so it always returns 0 exit code (it should work)
+    # Don't source mne_setup_sh here because changing PATH etc. can't be done in a script
     - if [ "${DEPS}" == "" ] && [ "${TRAVIS_OS_NAME}" == "linux" ]; then
         curl https://staff.washington.edu/larsoner/minimal_cmds.tar.gz | tar xz;
         export MNE_ROOT="${PWD}/minimal_cmds";

--- a/.travis.yml
+++ b/.travis.yml
@@ -80,11 +80,12 @@ matrix:
 before_install:
     - git clone https://github.com/astropy/ci-helpers.git
     - source ci-helpers/travis/setup_conda.sh
+    # Pipe the source into echo so it always returns 0 exit code (it should work)
     - if [ "${DEPS}" == "" ] && [ "${TRAVIS_OS_NAME}" == "linux" ]; then
         curl https://staff.washington.edu/larsoner/minimal_cmds.tar.gz | tar xz;
         export MNE_ROOT="${PWD}/minimal_cmds";
         export NEUROMAG2FT_ROOT="${PWD}/minimal_cmds/bin";
-        source ${MNE_ROOT}/bin/mne_setup_sh;
+        source ${MNE_ROOT}/bin/mne_setup_sh | echo;
       fi;
     - if [ "${TRAVIS_OS_NAME}" == "linux" ]; then
         /sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1400x900x24 -ac +extension GLX +render -noreset;

--- a/.travis.yml
+++ b/.travis.yml
@@ -80,12 +80,12 @@ matrix:
 before_install:
     - git clone https://github.com/astropy/ci-helpers.git
     - source ci-helpers/travis/setup_conda.sh
-    # Pipe the source into echo so it always returns 0 exit code (it should work)
+    # Pipe the source into tee so it always returns 0 exit code (it should work)
     - if [ "${DEPS}" == "" ] && [ "${TRAVIS_OS_NAME}" == "linux" ]; then
         curl https://staff.washington.edu/larsoner/minimal_cmds.tar.gz | tar xz;
         export MNE_ROOT="${PWD}/minimal_cmds";
         export NEUROMAG2FT_ROOT="${PWD}/minimal_cmds/bin";
-        source ${MNE_ROOT}/bin/mne_setup_sh | echo;
+        source ${MNE_ROOT}/bin/mne_setup_sh | tee;
       fi;
     - if [ "${TRAVIS_OS_NAME}" == "linux" ]; then
         /sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1400x900x24 -ac +extension GLX +render -noreset;

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,6 @@
 language: c
 sudo: false
+dist: precise
 cache:
   apt: true
   directories:

--- a/.travis.yml
+++ b/.travis.yml
@@ -86,7 +86,9 @@ before_install:
         curl https://staff.washington.edu/larsoner/minimal_cmds.tar.gz | tar xz;
         export MNE_ROOT="${PWD}/minimal_cmds";
         export NEUROMAG2FT_ROOT="${PWD}/minimal_cmds/bin";
-        source ${MNE_ROOT}/bin/mne_setup_sh | tee foo.txt;
+        export PATH=${MNE_ROOT}/bin:$PATH;
+        export LD_LIBRARY_PATH=${MNE_ROOT}/lib:$LD_LIBRARY_PATH;
+        mne_surf2bem --version;
       fi;
     - if [ "${TRAVIS_OS_NAME}" == "linux" ]; then
         /sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1400x900x24 -ac +extension GLX +render -noreset;

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,5 @@
 language: c
 sudo: false
-dist: precise
 cache:
   apt: true
   directories:
@@ -153,7 +152,7 @@ script:
         USE_DIRS="mne/";
       fi;
     - echo Testing with -a "${CONDITION}" ${COVERAGE} ${USE_DIRS}
-    - nosetests -a "${CONDITION}" -x --with-timer --timer-top-n 30 --timer-ok 5 --timer-warning 30 --verbosity=2 ${COVERAGE} ${USE_DIRS};
+    - nosetests -a "${CONDITION}" --with-timer --timer-top-n 30 --timer-ok 5 --timer-warning 30 --verbosity=2 ${COVERAGE} ${USE_DIRS};
     - if [ "${DEPS}" == "nodata" ]; then
         make pep;
         check-manifest --ignore doc,logo,mne/io/*/tests/data*,mne/io/tests/data,mne/preprocessing/tests/data;

--- a/doc/whats_new.rst
+++ b/doc/whats_new.rst
@@ -86,6 +86,8 @@ Changelog
 BUG
 ~~~
 
+    - Fixed a bug when creating spherical volumetric grid source spaces in :func:`setup_volume_source_space` by improving the minimum-distance computations, which in general will decrease the number of used source space points by `Eric Larson`_
+
     - Fix unit scaling when reading in EGI digitization files using :func:`mne.channels.read_dig_montage` by `Matt Boggess`_
 
     - Fix ``picks`` default in :meth:`mne.io.Raw.filter` to include ``ref_meg`` channels by default by `Eric Larson`_

--- a/mne/source_space.py
+++ b/mne/source_space.py
@@ -2672,7 +2672,7 @@ def _compare_source_spaces(src0, src1, mode='exact', nearest=True,
     if mode != 'exact' and 'approx' not in mode:  # 'nointerp' can be appended
         raise RuntimeError('unknown mode %s' % mode)
 
-    for s0, s1 in zip(src0, src1):
+    for si, (s0, s1) in enumerate(zip(src0, src1)):
         # first check the keys
         a, b = set(s0.keys()), set(s1.keys())
         assert_equal(a, b, str(a ^ b))
@@ -2725,10 +2725,11 @@ def _compare_source_spaces(src0, src1, mode='exact', nearest=True,
                     assert_true(len((s0['dist'] - s1['dist']).data) == 0)
         else:  # 'approx' in mode:
             # deal with vertno, inuse, and use_tris carefully
-            assert_array_equal(s0['vertno'], np.where(s0['inuse'])[0],
-                               'left hemisphere vertices')
-            assert_array_equal(s1['vertno'], np.where(s1['inuse'])[0],
-                               'right hemisphere vertices')
+            for ii, s in enumerate((s0, s1)):
+                assert_array_equal(s['vertno'], np.where(s['inuse'])[0],
+                                   'src%s[%s]["vertno"] != '
+                                   'np.where(src%s[%s]["inuse"])[0]'
+                                   % (ii, si, ii, si))
             assert_equal(len(s0['vertno']), len(s1['vertno']))
             agreement = np.mean(s0['inuse'] == s1['inuse'])
             assert_true(agreement >= 0.99, "%s < 0.99" % agreement)

--- a/mne/tests/test_dipole.py
+++ b/mne/tests/test_dipole.py
@@ -167,14 +167,15 @@ def test_dipole_fitting():
                                                      axis=1)))]
         amp_errs += [np.sqrt(np.mean((amp - d.amplitude) ** 2))]
         gofs += [np.mean(d.gof)]
-    assert_true(dists[0] >= dists[1], 'dists: %s' % dists)
-    assert_true(corrs[0] <= corrs[1], 'corrs: %s' % corrs)
-    assert_true(gc_dists[0] >= gc_dists[1] * 0.8,
+    factor = 0.8
+    assert_true(dists[0] / factor >= dists[1], 'dists: %s' % dists)
+    assert_true(corrs[0] * factor <= corrs[1], 'corrs: %s' % corrs)
+    assert_true(gc_dists[0] / factor >= gc_dists[1] * 0.8,
                 'gc-dists (ori): %s' % gc_dists)
-    assert_true(amp_errs[0] >= amp_errs[1],
+    assert_true(amp_errs[0] / factor >= amp_errs[1],
                 'amplitude errors: %s' % amp_errs)
     # This one is weird because our cov/sim/picking is weird
-    assert_true(gofs[0] <= gofs[1] * 2, 'gof: %s' % gofs)
+    assert_true(gofs[0] * factor <= gofs[1] * 2, 'gof: %s' % gofs)
 
 
 @testing.requires_testing_data

--- a/mne/tests/test_source_space.py
+++ b/mne/tests/test_source_space.py
@@ -254,7 +254,19 @@ def test_other_volume_source_spaces():
     src = read_source_spaces(temp_name)
     src_new = setup_volume_source_space(None, pos=7.0, mri=fname_mri,
                                         subjects_dir=subjects_dir)
-    _compare_source_spaces(src, src_new, mode='approx')
+    # we use a more accurate elimination criteria, so let's fix the MNE-C
+    # source space
+    assert_equal(len(src_new[0]['vertno']), 7497)
+    assert_equal(len(src), 1)
+    assert_equal(len(src_new), 1)
+    good_mask = np.in1d(src[0]['vertno'], src_new[0]['vertno'])
+    src[0]['inuse'][src[0]['vertno'][~good_mask]] = 0
+    assert_equal(src[0]['inuse'].sum(), 7497)
+    src[0]['vertno'] = src[0]['vertno'][good_mask]
+    assert_equal(len(src[0]['vertno']), 7497)
+    src[0]['nuse'] = len(src[0]['vertno'])
+    assert_equal(src[0]['nuse'], 7497)
+    _compare_source_spaces(src_new, src, mode='approx')
     assert_true('volume, shape' in repr(src))
     del src
     del src_new

